test/built-ins/Promise/prototype/finally/rejected-observable-then-calls-argument.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2019 Alexey Shvayka. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-catchfinallyfunctions
+description: >
+  thrower is anonymous built-in function with length of 1 that throws reason.
+info: |
+  Catch Finally Functions
+
+  ...
+  8. Let thrower be equivalent to a function that throws reason.
+  9. Return ? Invoke(promise, "then", « thrower »).
+
+  The "length" property of a Catch Finally function is 1.
+features: [Promise.prototype.finally, Reflect.construct]
+includes: [isConstructor.js]
+flags: [async]
+---*/
+
+Promise.reject(new Test262Error())
+  .finally(function() {})
+  .then($DONE, $DONE);
+
+var then = Promise.prototype.then;
+Promise.prototype.then = function(thrower) {
+  assert(!isConstructor(thrower));
+  assert.sameValue(thrower.length, 1);
```

Comment:
I can't find anywhere in the spec where this thrower function is defined to have a length of `1`.

Reply:
https://tc39.es/ecma262/#sec-thenfinallyfunctions and https://tc39.es/ecma262/#sec-catchfinallyfunctions